int32 encoder
---
bool success
